code/modules/mob/living/carbon/xenomorph/castes/spiderling/spiderling.dm

```diff
@@ -155,16 +164,25 @@
 
 /// resist when widow does
 /datum/ai_behavior/spiderling/proc/parent_resist()
+	SIGNAL_HANDLER
 	var/mob/living/carbon/xenomorph/spiderling/spiderling_parent = mob_parent
 	spiderling_parent.do_resist()
 
 /// rest and unrest when widow does
-/datum/ai_behavior/spiderling/proc/toggle_rest()
-	var/mob/living/carbon/xenomorph/spiderling/spiderling_parent = mob_parent
-	if(HAS_TRAIT(spiderling_parent, TRAIT_FLOORED))
-		spiderling_parent.set_resting(FALSE)
-	else
-		spiderling_parent.set_resting(TRUE)
+/datum/ai_behavior/spiderling/proc/start_resting(mob/source)
+	SIGNAL_HANDLER
+	var/mob/living/living = mob_parent
+	living.set_resting(TRUE)
+
+/datum/ai_behavior/spiderling/proc/stop_resting(mob/source)
+	SIGNAL_HANDLER
+	var/mob/living/living = mob_parent
+	living.set_resting(FALSE)
+	source.unbuckle_all_mobs()
```

Comment:
Did you mean to put this in start_resting?

Reply:
There's no reason to unbuckle the spiderlings until they get up, there's also the option of re-buckling them after the stat change, so there's no need for player re-attachment, what do you think? The reason we have to unbuckle is because this causes the widow to be trapped by the spiderlings until they un-attach the spiderlings from them